tests/kuery/test_parser.py

```diff
@@ -103,3 +103,10 @@ def test_optimization(self):
             "{'match': {'destination.ip': '169.254.169.254'}}]}}]}}"
         )
         self.assertEqual(dsl_str, good_case, "DSL string does not match the good case, optimization failed.")
+
+    def test_blank_space(self):
+        with self.assertRaises(kql.KqlParseError):
+            kql.lark_parse('"Test-ServiceDaclPermission" or"Update-ExeFunctions"')
```

Comment:
Add failure case with the space missing on the left side and failure case when `\n` is involved.

Reply:
Added in https://github.com/elastic/detection-rules/pull/3789/commits/1d8359c3b39bbb83e2281c8d92f02c77dfe2bf80